docs/CompVis_Stable_Diffusion_Instructions.md

```diff
@@ -0,0 +1,77 @@
+
+Once you have the account and an access token, authenticate yourself in a terminal or powershell console by running the following command.
+
+```
+huggingface-cli.exe login
+```
+
+It'll ask for your access token, which you can find on your account profile `Settings -> Access Tokens`, just copy it from here and carefully paste it on this prompt. Note that you won't see anything appear on the prompt when you paste it, that's fine. It's there already, just hit Enter. You'll start downloading the model from Hugging Face.
+
+## Converting to ONNX
+
+The model is trained with PyTorch so it can naturally convert to ONNX. Since we'll be using DirectML through ONNX Runtime, this step is needed. The script `convert_stable_diffusion_checkpoint_to_onnx.py` you already have here is just a local copy of the same file from the [Hugging Face diffusers GitHub repo](https://github.com/HuggingFace/diffusers/blob/main/scripts/convert_stable_diffusion_checkpoint_to_onnx.py), so you don't need to clone that repo.
+
+```
+python convert_stable_diffusion_checkpoint_to_onnx.py --model_path="CompVis/stable-diffusion-v1-4" --output_path="./stable_diffusion_onnx" --fp16
+```
+
+This will run the conversion and put the result ONNX files under the `stable_diffusion_onnx` folder. For better performance, we recommend you convert the model to half-precision floating point data type using the `--fp16` option.
+
+_Note: As of this writing, you cannot run this `--fp16` option until you have installed CUDA support with Torch packages using the following instruction. This will require up to 3 GB of additional disk space._
+
+```
+pip install torch>=1.13.0+cu116 torchvision>=0.13.0+cu116 torchaudio>=0.13.0 --extra-index-url https://download.pytorch.org/whl/cu116
+```
+
+## Running the ONNX Model
+
+You'll need a script that looks like what in the `text2image.py` file as follow. On a mid-range NVIDIA RTX 2070, a single image currently takes 20 seconds to generate from a prompt. It'll take up to 10 mins on the CPU.
+
+```python
+# (test/run.py)
+from diffusers import StableDiffusionOnnxPipeline
+pipe = StableDiffusionOnnxPipeline.from_pretrained("./stable_diffusion_onnx", provider="DmlExecutionProvider")
+prompt = "a photo of an astronaut riding a horse on mars."
+image = pipe(prompt).images[0]
+image.save("./result.png")
+```
+
+### A Debugging Note
+When running this script inside VSCode, the relative path specified here is relative to the base location of your project folder and not the location of your script file. To fix that up, configure the `cwd` (i.e. "current working directory") option in your launch.json file as follows:
+
+```json
+    // .vscode/launch.json
+    "configurations": [
+        {
+            "name": "Python: Current File",
+            "type": "python",
+            "request": "launch",
+            "program": "${file}",
+            "cwd": "${workspaceFolder}/test/",
+            "console": "integratedTerminal",
+            "justMyCode": true
+        }
```

Comment:
Curiously the lines below ("If you have an NVIDIA...") all appear in red on GitHub's diff view 🤔. I wonder if it's because of the missing `]` here?

Reply:
Yeah, that fixed it - looks normal now.

Reply:
Appeared fine to me, but see if putting in a closing bracket helps.